File: src/types/import.ts
import { Node, Edge } from ".";

export interface ImportRequest {
  strategy: "replace" | "merge";
  data: {
    version: string;
    exportDate: string;
    metadata: {
      nodeCount: number;
      edgeCount: number;
      appVersion?: string;
    };
    nodes: Node[];
    edges: Edge[];
  };
}

export interface ImportResponse {
  message: string;
  nodesCreated: number;
  edgesCreated: number;
  conflicts?: string[];
  warnings?: string[];
}

File: backend/models/import.go
package models

// Add this to your models package (models.go or similar):
type ImportData struct {
	Version    string                   `json:"version"`
	ExportDate string                   `json:"exportDate"`
	Metadata   ImportMetadata           `json:"metadata"`
	Nodes      []ReactFlowNode          `json:"nodes"`
	Edges      []map[string]interface{} `json:"edges"`
}

type ImportMetadata struct {
	NodeCount  int    `json:"nodeCount"`
	EdgeCount  int    `json:"edgeCount"`
	AppVersion string `json:"appVersion,omitempty"`
}
